index.ios.js

```diff
@@ -46,8 +56,14 @@
     /**
      * Executes event/command/jsFunction in webView.
      */
-    WebViewInterface.prototype._executeJS = function(strJSFunction){
-        return this.webView.ios.stringByEvaluatingJavaScriptFromString(strJSFunction);
+    WebViewInterface.prototype._executeJS = function(strJSFunction) {
+        return new Promise((resolve) => {
+            if (this.isUsingWKWebView) {
+                this.webView.ios.evaluateJavaScriptCompletionHandler(strJSFunction, (data) => resolve(data));
```

Comment:
Maybe you can also add error handling for this ?

Something like:

```
this.webView.ios.evaluateJavaScriptCompletionHandler(strJSFunction, (id, err) => {
    if (err) {
        return reject(err);
    }

   resolve(id);
});
```

Reply:
@surdu You are right, I could add this rejection and it would probably be a more clean approach anyways. I've also checked the signature of that method in the apple documentation and that would be possible. Anyways you would still encounter unhandled promise rejections as long as we don't change this line as well:

```js
this._executeJS('window.nsWebViewInterface._getIOSResponse('+oReqMsg.resId+')')
    .then(data => this._onWebViewEvent(eventName, data));
```

I would have to add a `.catch` block there as well and handle the error in order to actually make a difference. Unfortunately I feel a bit uncomfortable simply adding a `.catch(() => {})` because that's just muting any errors which might lead to debugging nightmares?

Please provide further information about the error you're getting here and how we could properly handle it. I have not had a single error so far thought.

Reply:
Well, you can throw the error from there, [as it was done here](https://github.com/shripalsoni04/nativescript-webview-interface/blob/master/index-common.js#L130).

Also, for the sake of consistency, maybe you could also switch arrow functions to normal inline functions as everywhere else in the code.

Reply:
That seems reasonable. I will do that later when I’m back on the computer. Thank you.

Reply:
@surdu Just pushed the changes and tested with 3.2 + 3.4. Still working great. 👍